src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkWatchListService.java

```diff
@@ -73,27 +84,72 @@ public void removeWatch(Endpoint endpoint) {
+        log.info("Discovery poll mode for endpoint {} update - exhausted mode requested (enable: '{}')",
+                endpoint, enable);
+        endpoints.computeIfPresent(endpoint,
+                (e, watchListEntry) -> watchListEntry.toBuilder().exhaustedPollEnabled(enable).build());
+    }
+
+    @VisibleForTesting
+    void updateAuxiliaryPoll(Endpoint endpoint, boolean enable, long currentTime) {
+        log.info("Discovery poll mode for endpoint {} update - auxiliary mode requested (enable: '{}')",
+                endpoint, enable);
+        WatchListEntry watchListEntry = endpoints.computeIfPresent(endpoint,
+                (e, listEntry) -> listEntry.toBuilder().auxiliaryPollEnabled(enable).build());
+
+        if (watchListEntry != null && !enable) { // when another mechanism used to determine ISL status is disabled
+            for (Set<Endpoint> e : timeouts.values()) {
+                e.remove(endpoint);
+            }
+
+            carrier.discoveryRequest(endpoint, currentTime);
+            addTimeouts(Collections.singleton(endpoint), currentTime + calculateTimeout(endpoint));
+        }
+    }
+
+    /**
+     * Handle update slow discovery flag request for BFD enabled case.
+     */
+    public void updateAuxiliaryPoll(Endpoint endpoint, boolean enable) {
+        updateAuxiliaryPoll(endpoint, enable, now());
+    }
+
     /**
      * Consume timer tick.
      */
-    public void tick(long tickTime) {
+    @VisibleForTesting
+    void tick(long tickTime) {
         SortedMap<Long, Set<Endpoint>> range = timeouts.subMap(0L, tickTime + 1);
         if (!range.isEmpty()) {
-            HashSet<Endpoint> renew = new HashSet<>();
+            Set<Endpoint> renewGeneric = new HashSet<>();
+            Set<Endpoint> renewExhausted = new HashSet<>();
+            Set<Endpoint> renewAuxiliary = new HashSet<>();
+
             for (Set<Endpoint> e : range.values()) {
                 for (Endpoint ee : e) {
-                    if (endpoints.contains(ee)) {
+                    WatchListEntry watchListEntry = endpoints.get(ee);
+                    if (watchListEntry != null) {
                         carrier.discoveryRequest(ee, tickTime);
-                        renew.add(ee);
+
+                        if (watchListEntry.isAuxiliaryPollEnabled()) {
+                            renewAuxiliary.add(ee);
+                        } else if (watchListEntry.isExhaustedPollEnabled()) {
+                            renewExhausted.add(ee);
+                        } else {
+                            renewGeneric.add(ee);
+                        }
                     }
                 }
             }
             range.clear();
-            if (!renew.isEmpty()) {
-                long key = tickTime + tickPeriod;
-                timeouts.computeIfAbsent(key, mappingFunction -> new HashSet<>())
-                        .addAll(renew);
-            }
+
+            addTimeouts(renewGeneric, tickTime + genericTickPeriod);
```

Comment:
You can reuse `calculateTimeout()` method here if you replace sets `renewGeneric`, `renewExhausted` and `renewAuxiliary` with map `Map<Endpoint, WatchListEntry> renew`. Benefits from here - single point of responsibility for delay calculation i.e. only `calculateTimeout()` will know how to do it and only it will need to be changed when the set of poll modes will be updated.

Reply:
Fixed.